src/OpenTelemetry.Instrumentation.AWSLambda/Implementation/AWSMessagingUtils.cs

```diff
@@ -127,12 +139,22 @@ internal static PropagationContext ExtractParentContext(SNSEvent.SNSMessage? mes
 
         var body = sqsMessage.Body;
         if (body != null &&
+#if NET6_0_OR_GREATER
+            body.TrimStart().StartsWith('{') &&
+            body.Contains(SnsMessageAttributes, StringComparison.Ordinal))
+#else
             body.TrimStart().StartsWith("{", StringComparison.Ordinal) &&
             body.Contains(SnsMessageAttributes))
+#endif
         {
             try
             {
-                snsMessage = JsonConvert.DeserializeObject<SNSEvent.SNSMessage>(body);
+#if NET6_0_OR_GREATER
+                using var stream = new MemoryStream(System.Text.Encoding.UTF8.GetBytes(body));
+                snsMessage = LambdaSerializer.Deserialize<SNSEvent.SNSMessage>(stream);
```

Comment:
I went this route as the Lambda serializer [has custom options to deal with the serialization](https://github.com/aws/aws-lambda-dotnet/blob/a2421013def1e938d24b3e9f9c3453868d77a360/Libraries/src/Amazon.Lambda.Serialization.SystemTextJson/AbstractLambdaJsonSerializer.cs#L129-L150), so I figured it was better to use that than just the vanilla source generator support. That requires the string to be turned into a `Stream` to use though.

Reply:
That seems overly complex. The custom options are not needed for the two properties we fetch, so maybe go with the simpler change like in #1270?

Reply:
I originally added this for fear of breaking the serialization after doing it the simpler way at first, but double-checking with the test, the easy way works fine, so I'll revert it back.